In [16]:
!pip install folium
import pandas as pd
import folium

In [17]:
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
data = pd.read_html(url)
data = data[0]
print(data)

    Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       

In [18]:
data.shape

(180, 3)

In [19]:
data.columns

Index(['Postal Code', 'Borough', 'Neighbourhood'], dtype='object')

In [20]:
data[data['Neighbourhood'] == 'Not assigned']

Postal Code       Borough Neighbourhood
0           M1A  Not assigned  Not assigned
1           M2A  Not assigned  Not assigned
7           M8A  Not assigned  Not assigned
10          M2B  Not assigned  Not assigned
15          M7B  Not assigned  Not assigned
..          ...           ...           ...
174         M4Z  Not assigned  Not assigned
175         M5Z  Not assigned  Not assigned
176         M6Z  Not assigned  Not assigned
177         M7Z  Not assigned  Not assigned
179         M9Z  Not assigned  Not assigned

[77 rows x 3 columns]

In [21]:
data2 = data[data['Borough'] != 'Not assigned']

In [22]:
data2.shape

(103, 3)

In [23]:
coorCsv = 'http://cocl.us/Geospatial_data'
Coords = pd.read_csv(coorCsv)
Coords

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

In [24]:
data2 = data2.sort_values(by = 'Postal Code')
data2 = data2.reset_index()[['Postal Code','Borough','Neighbourhood']]
data2

Postal Code      Borough  \
0           M1B  Scarborough   
1           M1C  Scarborough   
2           M1E  Scarborough   
3           M1G  Scarborough   
4           M1H  Scarborough   
..          ...          ...   
98          M9N         York   
99          M9P    Etobicoke   
100         M9R    Etobicoke   
101         M9V    Etobicoke   
102         M9W    Etobicoke   

                                         Neighbourhood  
0                                       Malvern, Rouge  
1               Rouge Hill, Port Union, Highland Creek  
2                    Guildwood, Morningside, West Hill  
3                                               Woburn  
4                                            Cedarbrae  
..                                                 ...  
98                                              Weston  
99                                           Westmount  
100  Kingsview Village, St. Phillips, Martin Grove ...  
101  South Steeles, Silverstone, Humbergate, Jamest...  
102                Northwest, West Humber - Clairville  

[103 rows x 3 columns]

In [25]:
Coords2 = Coords[['Latitude','Longitude']]
Coords2

Latitude  Longitude
0    43.806686 -79.194353
1    43.784535 -79.160497
2    43.763573 -79.188711
3    43.770992 -79.216917
4    43.773136 -79.239476
..         ...        ...
98   43.706876 -79.518188
99   43.696319 -79.532242
100  43.688905 -79.554724
101  43.739416 -79.588437
102  43.706748 -79.594054

[103 rows x 2 columns]

In [26]:
newData = pd.concat([data2,Coords2],axis = 1)
newData

Postal Code      Borough  \
0           M1B  Scarborough   
1           M1C  Scarborough   
2           M1E  Scarborough   
3           M1G  Scarborough   
4           M1H  Scarborough   
..          ...          ...   
98          M9N         York   
99          M9P    Etobicoke   
100         M9R    Etobicoke   
101         M9V    Etobicoke   
102         M9W    Etobicoke   

                                         Neighbourhood   Latitude  Longitude  
0                                       Malvern, Rouge  43.806686 -79.194353  
1               Rouge Hill, Port Union, Highland Creek  43.784535 -79.160497  
2                    Guildwood, Morningside, West Hill  43.763573 -79.188711  
3                                               Woburn  43.770992 -79.216917  
4                                            Cedarbrae  43.773136 -79.239476  
..                                                 ...        ...        ...  
98                                              Weston  43.706876 -79.518188  
99                                           Westmount  43.696319 -79.532242  
100  Kingsview Village, St. Phillips, Martin Grove ...  43.688905 -79.554724  
101  South Steeles, Silverstone, Humbergate, Jamest...  43.739416 -79.588437  
102                Northwest, West Humber - Clairville  43.706748 -79.594054  

[103 rows x 5 columns]

Let me tell you how i preprocessed the data. First i started with transforming data from wiki page. I extracted the tables at html page. Web page returned list of tables and our data was at first index of list. The table has 180 rows and 3 columns. I dropped columns that have 'Not assigned' value at Borough column. This method dropped 77 rows from dataset. I didn't apply any other method to preproccess data and our new dataset has 103 rows and 3 columns. Because of i don't want to deal with unreliable package. I used the other csv file that provided by IBM.  I reordered our neighbor dataset by Postal Code column and merged two files into one. Thats all i did before i continue.

In [27]:
latitude = 43.72534265655438
longitude = -79.37259628709921

I used borough column to draw points at map.

In [28]:
map = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(newData['Latitude'], newData['Longitude'], newData['Borough']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map)  
    
map